java/memory/memory-core/pom.xml

```diff
@@ -33,6 +33,7 @@ under the License.
   <properties>
     <!-- List of add-opens arg line arguments for this module's tests -->
     <surefire.add-opens.argLine>--add-reads=org.apache.arrow.memory.core=ch.qos.logback.classic --add-opens=java.base/java.lang.reflect=org.apache.arrow.memory.core --add-opens=java.base/java.nio=org.apache.arrow.memory.core,ALL-UNNAMED</surefire.add-opens.argLine>
+    <arrow.memory.debug.allocator>true</arrow.memory.debug.allocator>
```

Comment:
@lidavidm adding this property and manually enables the logs, though a test is failing https://github.com/apache/arrow/actions/runs/10614222183/job/29419683887?pr=43880#step:6:3553 

I think we need to `assertTrue` this if we enable debug logs. And probably need to pick the property from the code. It should be okay?

Reply:
By the way, I tried to set up an environmental variable for the pom.xml and set it via the build script. Would that be a better approach or should we just ask users to use true or false in the pom.xml?